### DISCUSSION TITLE
Symbolic links

### DISCUSSION THREAD

**Discussion by mrjrozycki** [mar10/wsgidav#287] [General]
Hi everyone
Does wsgidav support symbolic links? We are working on a project where we want to serve files with wsgidav but serving symbolic links is pretty important to us. Is it supported in this library?
I already tried creating a small server, and it seems that symlinks were just replaced with files they link to, which is not ideal for us. Maybe I did something wrong tho and that is why I am asking :)

**Comment by mar10** [comment]
Could you explain in a bit more detail, what you are trying to achieve:
- Are you linking files or directories?  
- How would we handle identical resources that are linked multiple times (Locking is is based on the resource URL).
- How would you expect the URL to change address bar of the browser when clicking a folder in the directory browser.

There is a [stale RFC4437](https://greenbytes.de/tech/webdav/rfc4437.xhtml) related to this. I don't think it's implemented by many clients though.
The general problem of multiple occurrences of a single resource, could also be handled by a [custom middleware (example)](https://github.com/mar10/wsgidav/blob/master/wsgidav/samples/virtual_dav_provider.py).

FWIW:
with 4.3.0 a new option is introduced to control the handling of symlinks:
Note that `fs_dav_provider.follow_symlinks` is false by default, since [symlinks may be a security risk](https://serverfault.com/questions/244592/followsymlinks-on-apache-why-is-it-a-security-risk).

```yaml
#: Addional configuration passed to `FilesystemProvider(..., fs_opts)`
fs_dav_provider:
    #: Serve symbolic link files and folders
    follow_symlinks: true
```
